cpp/harvest/nacjd.cc

```diff
@@ -158,6 +171,32 @@ void ExtractExistingIDsFromMarc(MARC::Reader * const marc_reader, std::set<std::
     }
 }
 
+void ExtractOneIDFromWebsite(const unsigned nacjd_id, const std::string nacjd_path) {
+    std::string nacjd_title_path(nacjd_path + NACJD_TITLES_FILE_NAME);
+    std::string nacjd_JSON_path(nacjd_path + NACJD_NEW_TITLES_JSON_FILE_NAME);
+
+    if (FileUtil::Exists(nacjd_title_path))
+        FileUtil::DeleteFile(nacjd_title_path);
+
+    FileUtil::WriteString(nacjd_title_path, "");
+
+    std::ifstream file(nacjd_title_path);
+    if (file)
+        std::for_each(std::istream_iterator<char>(file), std::istream_iterator<char>(), HandleChar);
+    else
+        LOG_ERROR("couldn't open file: " + nacjd_title_path);
+    if (FileUtil::Exists(nacjd_JSON_path) and not FileUtil::DeleteFile(nacjd_JSON_path))
+        LOG_ERROR("Could not delete file: " + nacjd_JSON_path);
+
+    std::ofstream json_new_titles(nacjd_JSON_path);
+
+    json_new_titles << "{ \"nacjd\" : [ " << '\n';
```

Comment:
I wonder if we should use a different JSON wrapper here, but we might leave that open for now and discuss it for a later PR when we add the other modes.